test/distributed/test_c10d_pypg.py

```diff
@@ -156,6 +156,15 @@ def test_ddp_with_pypg_with_grad_views(self):
             pg, [torch.device("cpu")], device_ids=None, gradient_as_bucket_view=True
         )
 
+    def test_ddp_no_init_sync(self):
+        pg = self._get_process_group()
+
+        model = nn.Sequential(nn.Linear(2, 2), nn.ReLU())
+        model = DDP(model, process_group=pg, init_sync=False)
```

Comment:
just curious, if rank0 has DDP(model0) and rank1 has DDP(model1), both with `init_sync=False`, when does the error get raised / what type of error is it?

Reply:
I believe there will be a deadlock as they are training a different model.

Reply:
That doesn't sound good, could we somehow run a shape check in the first backward pass or similar in that case, or at least add the disclaimer in the docs for `init_sync`?

Reply:
Nvm on the docs part, it looks like the disclaimer is there!